### DISCUSSION TITLE
Custom priors in LightweightMMM

### DISCUSSION THREAD

**Discussion by cirolista** [google/meridian#349] [General]
**Context**
In our business, we have some channels that exhibit an on-off behavior in terms of spending and have limited data points. For these channels, we observe a very slow degradation of contributions. 

**Proposed Solution**
To mitigate this, the idea is to adjust the prior related to the "lag_weight".

**Code**
custom_priors={
             "lag_weight": numpyro.distributions.Beta(concentration1=jnp.array([1,1,1,1,1,1,1,1]), concentration0=jnp.array([2,2,2,2,2,2,10,2]))
             }

The code works just fine, resulting in a rapid decay of contributions. However, during further testing, we encountered an unexpected behavior. Enabling 64-bit precision (jax.config.update("jax_enable_x64", True)) leads to slight variations in the results, even when the model configuration remains unchanged (i.e., without custom priors for individual channels). We are seeking to understand the potential reasons behind this observation.

**Code**

custom_priors={
             "lag_weight":numpyro.distributions.Beta(concentration1=1, concentration0=2), }

**Comment by brendalewisprice** [comment]
Thank you for reaching out for support. This is the discussion page for Google's new MMM product Meridian, so I suggest you reach out to the [Lightweight MMM Discussion Page](https://github.com/google/lightweight_mmm/discussions) for help with your LMMM related question.

**Comment by dirknbr** [comment]
Agree we need to discuss this on LMMM forum. Please link the LMMM issue link here if you have it.

Please note we have extremely limited support for LMMM while Meridian is being launched.

Can you please elaborate on the 'slight variation' you mention.

Potentially the jnp arrays need to be floats and not int.

Dirk
